Browser agnostic Baseline site & content

petele opened the pull request:
Most of the content was written by @rachelandrew and is pulled from https://web.dev/baseline and made to be browser agnostic. This content is not meant to replace the [`/docs`](https://github.com/web-platform-dx/web-features/tree/main/docs) folder, but to create a landing page that we can point web developers to that explains What is Baseline, How to use Baseline, and makes the Baseline icons and logos available for their use.

You can see the current draft content at https://petele.github.io/baseline-landing/

This uses/d the default [Minimal](https://github.com/pages-themes/minimal) theme, with updates from @meyerweb to use CSS Grid, RWD, and font-size relativity.

petele commented on gh-pages/baseline-in-the-wild.md:
```diff
@@ -0,0 +1,24 @@
+
+- [Baseline on MDN (Firefox)](https://developer.mozilla.org/en-US/blog/baseline-unified-view-stable-web-features/)
+- [Baseline on web.dev (Google)](https://web.dev/baseline/)
+<!-- - Baseline on MSDN (Microsoft) -- Couldn't find appropriate link -->
```
@captainbrosset - is there a good site/page I should like to for Microsoft/Edge?

captainbrosset replied:
No there isn't a Microsoft Edge page about Baseline for now.

meyerweb commented:
I tried to fork in order to make some CSS upgrades and could not for the life of me figure out how to get `gh-pages` to come with the fork, or be created after I forked, which kept me from GH Paging the preview site to work on and then submitting a pull request.

So… how would you like me to suggest the CSS upgrades?  (Pointers to how to make the Pages preview work in a fork are welcome too.)

petele commented:
> I tried to fork in order to make some CSS upgrades and could not for the life of me figure out how to get `gh-pages` to come with the fork

Ugh, yay. I think you may have to fork my fork, then go from there, which sounds like way more of a pain than it should be. To make it easier, if you want to either send the suggestions as comment, or email them to me, I can make that work. Or if you want to send me an updated file(s) by email, that works too.

I'll also add a readme to this folder with more info about how to build locally, which should work once this has been merged.

meyerweb commented:
I tried to fork your fork, but maybe I did it wrong?  I dunno man, git is a land of infinite mystery to me.

Anyway, this is what I came up with to change or add so far (comments indicate things to remove outright):

```
.wrapper {
  width: 61.5em;
  display: grid;
  grid-template-columns: 19.25em auto;
  gap: 0 6.5em;
}

header {
  /* float: left; */
  width: 19.25em;
  grid-column: 1;
}

section {
  /* float: right; */
  grid-column: 2;
  padding-block-start: 1.33em;
}
```

It’ll need some responsiveness but let’s see what those changes look like first.

petele commented:
@meyerweb - I've added instructions for building locally. You'll need to clone my repo, then follow the instructions in the README.md should work.

In the mean time, I'll add the changes you just made.

petele commented:
@meyerweb - 100% agree on Git, there are days where I just delete my branch and start over. Sigh. One thing, you'll want to use the `pages` branch, not the `gh-pages` branch, which confusingly also exists in my repo.

meyerweb commented:
incredibly, it still doesn’t work (not that I blame you, just letting you know).  I went to [petele:pages](https://github.com/petele/web-features/tree/pages) and cloned it locally via `git clone git@github.com:petele/web-features.git`.  Here’s the directory listing of what I got, showing a complete lack of `/gh-pages` and its various resources:

```
drwxr-xr-x   29 emeyer  staff     928 Aug 29 16:30 .
drwxr-xr-x   30 emeyer  staff     960 Aug 29 16:30 ..
drwxr-xr-x   13 emeyer  staff     416 Aug 29 16:30 .git
-rw-r--r--    1 emeyer  staff      43 Aug 29 16:30 .gitattributes
drwxr-xr-x    6 emeyer  staff     192 Aug 29 16:30 .github
-rw-r--r--    1 emeyer  staff     243 Aug 29 16:30 .gitignore
-rw-r--r--    1 emeyer  staff       3 Aug 29 16:30 .node-version
-rw-r--r--    1 emeyer  staff       6 Aug 29 16:30 .nvmrc
-rw-r--r--    1 emeyer  staff     329 Aug 29 16:30 .prettierignore
-rw-r--r--    1 emeyer  staff       3 Aug 29 16:30 .prettierrc
drwxr-xr-x    3 emeyer  staff      96 Aug 29 16:30 .vscode
-rw-r--r--    1 emeyer  staff    5899 Aug 29 16:30 2022-backgrounder.md
-rw-r--r--    1 emeyer  staff   16532 Aug 29 16:30 GOVERNANCE.md
-rw-r--r--    1 emeyer  staff   11357 Aug 29 16:30 LICENSE.txt
-rw-r--r--    1 emeyer  staff    1941 Aug 29 16:30 README.md
drwxr-xr-x    7 emeyer  staff     224 Aug 29 16:30 docs
-rw-r--r--    1 emeyer  staff     674 Aug 29 16:30 eslint.config.js
drwxr-xr-x  925 emeyer  staff   29600 Aug 29 16:30 features
drwxr-xr-x   45 emeyer  staff    1440 Aug 29 16:30 groups
-rw-r--r--    1 emeyer  staff    7522 Aug 29 16:30 index.ts
-rw-r--r--    1 emeyer  staff  151313 Aug 29 16:30 package-lock.json
-rw-r--r--    1 emeyer  staff    2321 Aug 29 16:30 package.json
drwxr-xr-x    4 emeyer  staff     128 Aug 29 16:30 packages
drwxr-xr-x    3 emeyer  staff      96 Aug 29 16:30 schemas
drwxr-xr-x   13 emeyer  staff     416 Aug 29 16:30 scripts
drwxr-xr-x   13 emeyer  staff     416 Aug 29 16:30 snapshots
-rw-r--r--    1 emeyer  staff   19676 Aug 29 16:30 towards-features.md
-rw-r--r--    1 emeyer  staff     221 Aug 29 16:30 tsconfig.json
-rw-r--r--    1 emeyer  staff    2859 Aug 29 16:30 types.ts
```

So the Gemfile’s missing, which means `bundle` has nothing to work with, etc., etc.  It weirds me out that whether I fork or clone your repository, that directory just doesn’t make it over!

petele commented:
Did you do a `git checkout pages`?  I think you need to do that in order to switch to that branch once you've cloned the repo.

meyerweb commented:
> Did you do a `git checkout pages`? I think you need to do that in order to switch to that branch once you've cloned the repo. Then, you should see the `/gh-pages/` folder.

DOH!  I hereby grant myself ALL the facepalms.  Thank you.

captainbrosset commented:
I really like the content. Thank you for getting this started.

A general comment I have about the PR though, is that it comes with a lot of new files, more than I would have expected. Seems like more code than we really need for a simple static site.

I'm not a huge fan of using Ruby/Gems/Jekyll here as it introduces a new dependency. Maintainers of this repo already have node/npm (since it's how we manage deps and build), so I would have preferred to stick with this if possible (eleventy seems like a good choice here for example).

Also, do we need to load a font file? I happened to have Noto Sans installed locally, so there's no perf hit for me. But, seeing how close to a system UI font it looks, and that our website is a simple information one, where design isn't super important, I would ask that we consider just using a system font instead.

Also, do we need to use Sass here? Given that we document the web platform, I would love it if we did an effort to use the platform instead of libraries. Scanning quickly through the code, I don't see obvious usage of Sass features here. Seems like we could just use CSS straight up.

Finally, the theme we use seems to be coming with a lot of CSS code, which, seeing the rendered site, feels surprising. I wouldn't expect us to need ~600 lines of CSS code for our content, which is mostly headers, paragraphs, and links, and a bit of complexity for the menu. (the Coverage tool in DevTools shows me 86% of unused CSS in style.css). Can we replace with a single CSS file with just the styles we need?

petele commented:
@captainbrosset - I completely understand your concern. As you mentioned, the site is based on [GitHub pages and uses Jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll), which is provided by the GitHub infrastructure, and includes a set of [pre-defined themes/templates](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/adding-a-theme-to-your-github-pages-site-using-jekyll). That reduces the number of dependancies we need to keep up to date, and made creating the site much easier. To get something out somewhat quickly, I used the [minimal theme](https://github.com/pages-themes/minimal) which is provided by the infrastructure. I've used eleventy before, and yes, it is a good option, but would have had a higher complexity level for build/deployment, maintenance, and so forth.  I felt this was faster, and there were less potential things to break.

> Also, do we need to load a font file?

This came from the theme, I'm happy to take PRs if you want to do the work to pull it out. I did add `font-display: swap` to ensure that there is no rendering hit if the font isn't available and needs to be loaded. 

> Also, do we need to use Sass here?

It comes for free as part of the GitHub Pages infrastructure, and was used as part of the template. 

> theme we use seems to be coming with a lot of CSS code

There are a lot of things in the template that could be cleaned up, we're not using a lot of the styling provided, but I kept it in so it was there if needed in the future. Most likely one of the biggest is the unused pieces is the code highlighting functionality, which I do suspect we'll want to include at some point, and taking that out now will cause headaches later. And even without cleaning up the (S)CSS, the entire landing page is about 30k, with the CSS file landing at only 2.2k.

For this PR, I focused on getting something up that looks good and is functional today, with as much flexibility for the future as possible.

ddbeck commented:
I'm going to start on an actual content review after this but first some preliminaries:

I agree with @captainbrosset that the new dependencies are a problem and spooked me a bit, as the person reviewing nearly all the Dependabot PRs of late. I am OK with having it as a short-term thing, provided we have at least a vague plan for what follows. I know GitHub sort of blesses Jekyll as a way to do this, but the context switch and (at least in my experience on macOS) frequent local development breakages makes the tools somewhat unattractive to me.

One way to cope with this: if someone is willing to commit to reviewing such a PR (or a series of PRs), I'd be willing to port this over to something a bit more familiar (like [Astro](https://astro.build/) or the docs-specific [Starlight](https://starlight.astro.build/)) after we've landed the initial Jekyll site. If we're not committed to pixel-perfect equivalence, then we could probably get it done fairly quickly using some different Lego bricks.

In the mean time, I'll have a look at the content. I think it's important that we have something like this, so I won't hold up merging on the tools per se, but I do want to have some idea of where we're headed when we do land this.

ddbeck commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,37 @@
+permalink: web-features
+---
+
+# Web features
```
```suggestion
# web-features
```

ddbeck commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,37 @@
+
+For example, the `fetch()` API feature is described in the [features/fetch.yml](https://github.com/web-platform-dx/web-features/blob/main/features/fetch.yml) file.
+
+In addition to meta information about the feature, the various elements that come together to make the feature are identified using a list of [browser-compat-data](https://github.com/mdn/browser-compat-data) entries, from the MDN repository. This repository already holds fine-grained support data for most web platform features, this is used to build up each feature grouping, by allowing us to ide ...
```
Some terminology adjustments and avoiding suggesting that we use the BCD repo directly, as opposed to the published package. Avoiding the ambiguous first person plural.

```suggestion
In addition to meta information about the feature, the various parts that come together to make the feature are identified using [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) entries. BCD holds fine-grained support data for most web platform features; this is used to build up each feature definition, identifying all of the component parts.
```

ddbeck commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,37 @@
+
+See [What makes a feature](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md#what-makes-a-feature) for more details.
+
+## How are features used?
+
+The web features information is the basis that allows us to identify Baseline features, therefore anywhere that displays Baseline status should use this repository as the source of truth.
```
There are other applications for feature definitions. I think narrowing the scope of the question makes this easier to write and edit:

```suggestion
## How does web-features relate to Baseline?

The web-features project computes, reviews, and publishes Baseline statues. Therefore, web-features is the source of truth for Baseline statuses.
```

ddbeck commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,37 @@
+
+## Get involved
+
+It’s a huge effort to group the entire web platform in this way, and will be an ongoing one given that new web platform features are released every month. The community is encouraged to help with the grouping work. See the [contributing information](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md) for more details.
```
Avoiding the "group" terminology since we have groups and features and it's an occasional point of confusion. I'm also suggesting an explicit call to action.

```suggestion
It’s a huge effort to describe the entire web platform in this way and will be an ongoing one given that new web platform features are released every month. Help out by contributing feature definitions. See the [contributing information](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md) for more details.
```

ddbeck commented on gh-pages/webdx-cg.md:
```diff
@@ -0,0 +1,17 @@
+
+> The mission of the [Web Developer Experience (WebDX) Community Group](https://www.w3.org/community/webdx/) is to facilitate coordinated approaches to improve the overall experience of developing for the Web platform when such coordination provides unique opportunities for these improvements. — [WebDX Community Group charter](https://github.com/web-platform-dx/admin/blob/main/charter.md).
+
+A subset of the members of the CG make up the owner and governance body for Baseline. The official definition of what Baseline is, including the goals and non-goals of the project, can be found in the [web-platform-dx/web-features repository](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md).
+
+The community group also owns the [web feature grouping work](web-features).
```
The Venn diagram with WebDX is kinda complicated. I suggest mostly evading this with something like:

```suggestion
The web-features project grew out of the work of the WebDX Community Group. WebDX plays an important part in coordinating the web-features project by hosting meetings and making connections to research, standardization work, and other efforts.
```

ddbeck reviewed (changes requested):
OK, a bunch of comments but a lot of them are nits. The non-style nits have to do with:

- "Ready to use" and "safe to use" have been controversial but avoidable; we have good alternatives and I've tried to suggest them. There's some similar issues describing Baseline, with suggestions to reuse the already-scrutinized language on MDN.
- We/us is ambiguous without bylines. I've made some guesses about how to avoid them, or make authorship explicit.
- The web-features ⇔ WebDX relationship isn't as tidy as one might guess. I suggested some ways to dodge that.

captainbrosset commented:
At the risk of being the naysayer here, I don't see the value of merging a somewhat large Jekyll-based infra quickly here, fully knowing that we don't want to support it later and want to change it for something else soon.

To me, the right thing to do in this PR is to focus on the actual content (which is great, and which Daniel has made a bunch of comments about), put it into the /docs/ folder for now, and point people to that. My recollection is that github_pages can be setup to serve *.md files located under the /docs/ folder with just one configuration change in the repo, and no additional files.

I agree with the content direction and the stated goal to own the Baseline landing page. I just don't see the urgency for merging this PR as is.

captainbrosset commented on gh-pages/index.md:
```diff
@@ -0,0 +1,33 @@
+# What is Baseline?
+
+Web Platform Baseline gives you clear information about which web platform features are ready to use in your projects today. When reading an article, or choosing a library for your project, if the features used are all part of Baseline, you can trust the level of browser compatibility. By aligning with Baseline, there should be no surprises when testing your site.
```
I like Daniel's suggested changes. We don't want to take shortcuts and say that Baseline is a guaranty. Every project is different, every site has different users and users' devices. Baseline doesn't take all browsers into considerations, and doesn't take all aspects of a feature into consideration either. And the 30 months delay for Baseline high is a choice not every project will accept.

I would go a bit further though, and mention that the "browsers" are only the ones we list in the next section.

```suggestion
Baseline gives you clear information about which web platform features work across our [core browser set](link to next section) today. When reading an article, or choosing a library for your project, if the features used are all part of Baseline, you can trust the level of browser compatibility. By aligning with Baseline, there should be fewer surprises when testing your site.
```

petele replied:
manually fixed in separate checkin.

captainbrosset commented on gh-pages/index.md:
```diff
@@ -0,0 +1,33 @@
+
+## How do features become part of Baseline?
+
+Features that are in Baseline are interoperable across a core browser set:
```
```suggestion
Features that are in Baseline are interoperable across the following core browser set:
```

captainbrosset commented on gh-pages/index.md:
```diff
@@ -0,0 +1,33 @@
+Features that are in Baseline are interoperable across a core browser set:
+
+- Chrome (desktop and Android)
+- Edge
```
Edge also exists on mobile, so it's worth adding `(desktop)` on this line.

petele replied:
fixed

captainbrosset commented on gh-pages/index.md:
```diff
@@ -0,0 +1,33 @@
+
+## How do features become part of Baseline?
+
+Features that are in Baseline are interoperable across a core browser set:
+
+- Chrome (desktop and Android)
+- Edge
+- Firefox (desktop and Android)
+- Safari (macOS and iOS)
+
+Baseline has two stages:
+
+- **Newly available**: The feature becomes supported by all of the core browsers, and is therefore interoperable. Indicated with a blue icon.
+- **Widely available**: 30 months has passed since the newly interoperable date. The feature can be used by most sites without worrying about support. Indicated with a green icon.
+
+Features that are newly available can be used with confidence that they work in all the core browsers, however most people will need to consider fallbacks for browsers without support.
+
+Features that are widely available should be considered broadly available for use.
```
+1

captainbrosset commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,37 @@
+
+For example, the `fetch()` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.
+
+## How is a feature described?
```
I would actually remove this section. As a casual reader, I'm not interested in these internal details. I'm also not interested in this as a consumer of the package. If I'm looking to contribute, then yes, but in that case I would much rather go and read the contributor docs on the repo directly.

petele replied:
removed it

captainbrosset commented on gh-pages/webdx-cg.md:
```diff
@@ -0,0 +1,17 @@
+
+# The WebDX Community Group
+
+> The mission of the [Web Developer Experience (WebDX) Community Group](https://www.w3.org/community/webdx/) is to facilitate coordinated approaches to improve the overall experience of developing for the Web platform when such coordination provides unique opportunities for these improvements. — [WebDX Community Group charter](https://github.com/web-platform-dx/admin/blob/main/charter.md).
+
+A subset of the members of the CG make up the owner and governance body for Baseline. The official definition of what Baseline is, including the goals and non-goals of the project, can be found in the [web-platform-dx/web-features repository](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md).
+
+The community group also owns the [web feature grouping work](web-features).
```
I would actually simplify the intro even more:

```suggestion
The [Web Developer Experience (WebDX) Community Group](https://www.w3.org/community/webdx/) is a W3C Community Group which coordinates the work on [web-features](/web-features) and [Baseline](/baseline). For more information, see the group's [charter](https://github.com/web-platform-dx/admin/blob/main/charter.md).
```

petele replied:
Going to stick with @ddbeck suggestion for now. Feel free to reopen if you feel strongly.

petele commented on gh-pages/name-and-logo-usage-guidelines.md:
```diff
@@ -0,0 +1,152 @@
+  </tr>
+</table>
+
+The Baseline Widely Available logo should be used when talking about Baseline as a concept, or to indicate a specific web platform feature has been supported by all of the core browsers for at least 30 months, and is therefore widely interoperable.
```
The definition we're using on the landing page is 2.5 years or 30 months, we should be consistent. We generally don't ask developers to look at the web-features package data, and I don't want to do that to them here.

ddbeck replied:
> We generally don't ask developers to look at the web-features package data

Why not? It's way easier than actually following the [formal definition](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md#interoperable-low-status) from scratch. It contains a number of catches and exceptions.

This combined with https://github.com/web-platform-dx/web-features/pull/1706/files#r1743829760 and https://github.com/web-platform-dx/web-features/pull/1706/files#r1743836903 worries me a bit. If this file isn't really under the control of web-features, then I think it might be better if one of these things happen:

- Google hosts the legalese elsewhere and we link to it. If we're not free to change it after it merges, then that would be more honest and transparent about the authorship and maintenance of the usage guidelines. We could have some sort of cheat sheet or implementers guide (that's more specifically about integrating against web-features), that links out to the legalese. 

- The text here changes to say something like "The Baseline Widely Available logo should be used when talking about Baseline as a concept, or to indicate a specific web platform feature has satisfied the status requirements described in [the Baseline status definition](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md#wider-support-high-status)."

petele commented on gh-pages/name-and-logo-usage-guidelines.md:
```diff
@@ -0,0 +1,152 @@
+
+### How are the logos licensed?
+
+The logos have been released under the [Creative Commons Attribution No Derivatives 4.0 License](https://creativecommons.org/licenses/by-nd/4.0/). However, since the Baseline name and logos are trademarks owned by Google, we have prepared these trademark usage guidelines to describe their appropriate uses.
```
Will check with lawyers.

petele commented on gh-pages/baseline-in-the-wild.md:
```diff
@@ -0,0 +1,28 @@
+
+## From browser vendors
+
+- [Baseline on MDN (Firefox)](https://developer.mozilla.org/en-US/blog/baseline-unified-view-stable-web-features/)
```
I changed the title to `Features projects and tools`, and added the web-features explorer.

I do want to keep a link to the browser vendors up top, I think it's an important part to link to their content. I've also updated the link on MDN, and am happy to use something else if you have suggestions.

petele commented on gh-pages/use-baseline.md:
```diff
@@ -0,0 +1,22 @@
+
+Writing a blog post or article? Or presenting at a conference? Use the [Baseline logos](name-and-logo-usage-guidelines) as a shorthand to let your audience know the status of a feature.
+
+The [Universal Baseline Widget](https://github.com/GoogleChrome/baseline-status) is also available to include on your site or pages to indicate the status of a web feature. It renders at run-time, so it’s always up to date with the latest Baseline status.
```
added

petele commented:
@captainbrosset - if I switch this to Eleventy, will that get you on board?

captainbrosset commented:
> @ddbeck & @captainbrosset - if I switch this to Eleventy, will that get you on board? I think I can make the change(s) within about an hour or so...

For my part, yes.

ddbeck commented on gh-pages/name-and-logo-usage-guidelines.md:
```diff
@@ -0,0 +1,152 @@
+
+[Web Platform Baseline](./) is an initiative that gives you clear information about which web platform features are ready to use in your projects today. When reading an article, or choosing a library for your project, if the features used are all part of Baseline, you can trust the level of browser compatibility.
+
+The Baseline name and logos (“Baseline Trademarks”) are trademarks owned by Google, but have been licensed so they can be used by third parties to indicate which web platform features work well across all browsers. We've put together these guidelines to promote the growth of the Baseline initiative while making sure that the “Baseline” brand is protected as a meaningful source identifier in a way ...
+
+We encourage you to use the Baseline name and logos in articles, presentations, sites, or anywhere that you’re talking about web platform features, to help developers understand browser support for those features.
```
See also https://github.com/web-platform-dx/web-features/pull/1706/files#r1745282391

petele commented on gh-pages/name-and-logo-usage-guidelines.md:
```diff
@@ -0,0 +1,152 @@
+### Where can I check the status of a web platform feature?
+
+Baseline is already displayed for many features on [Can I Use](https://caniuse.com/), [MDN](https://developer.mozilla.org/), and all features on [webstatus.dev](https://webstatus.dev) so you can see at a glance the status of features. We hope that many other sites will start to use Baseline to clearly explain feature status—from individual blogs to large content sites. Find out how to [use Baseli ...
+
```
removed.

tidoust commented on gh-pages/_layouts/default.html:
```diff
@@ -0,0 +1,52 @@
+      <header>
+        {% if site.logo %}
+        <a href="./">
+          <img src="{{site.logo | relative_url}}" alt="Logo" />
```
The alt text should probably be "Baseline".

petele commented on packages/web-features/tsconfig.json:
```diff
@@ -2,6 +2,7 @@
     "module": "ESNext",
-    "moduleResolution": "Bundler"
+    "moduleResolution": "Bundler",
+    "skipLibCheck": true
```
Needed to add this because it was throwing a build error on build due to an issue with the `markdown-it` module. Adding this tells it not to get cranky if some of the node modules aren't easily compatible with ESM

Error was:

```
../../node_modules/@types/markdown-it/dist/index.cjs.d.ts:981:23 - error TS2709: Cannot use namespace 'LinkifyIt' as a type
```

ddbeck replied:
This is a very blunt instrument to accommodate Eleventy's lack of ESM support—I'm not comfortable with it. I think a better way to fix this would be:

- Move `gh-pages` to `/gh-pages/src`
- Move all the root Eleventy config to `/gh-pages` (i.e., `./eleventy.config.js` → `gh-pages/eleventy.config.js`, move the `scripts` changes from `package.json` into `gh-pages/package.json`)

Then we can run gh-pages-related things separate from everything else, instead of relaxing the type checks on the code that generates the web-features data.

petele replied:
Done

petele commented on tsconfig.json:
```diff
@@ -5,7 +5,8 @@
     "resolveJsonModule": true,
-    "noEmit": true
+    "noEmit": true,
+    "skipLibCheck": true
```
Needed to add this because it was throwing a build error on build due to an issue with the `markdown-it` module. Adding this tells it not to get cranky if some of the node modules aren't easily compatible with ESM

Error was:

```
../../node_modules/@types/markdown-it/dist/index.cjs.d.ts:981:23 - error TS2709: Cannot use namespace 'LinkifyIt' as a type
```

petele replied:
Removed in 167f11cab74d12c55501e35fa3be9ff6664c95e2

petele commented:
I've moved this to use 11ty, and removed the SCSS dependancy. Adding 11ty added 3 new packages and required some changes to the `tsconfig.json` files so that they ignore node modules. 

I've integrated all of the suggestions & changes from @ddbeck and @captainbrosset. 

I believe we should be good to go now.

captainbrosset commented on gh-pages/web-features.md:
```diff
@@ -0,0 +1,26 @@
+
+## How does web-features relate to Baseline?
+
+The web-features project computes, reviews, and publishes Baseline statues. Therefore, web-features is the source of truth for Baseline statuses.
```
```suggestion
The web-features project computes, reviews, and publishes Baseline statuses. Therefore, web-features is the source of truth for Baseline statuses.
```

captainbrosset reviewed (approved):
The content seems good to me, and I like the general direction of using 11ty and simplifying the CSS. Approving, but I know Daniel will likely have more comments about the technical aspects of this.
Thank you for the PR!

atopal commented on gh-pages/index.md:
```diff
@@ -0,0 +1,35 @@
+
+## How do I find the Baseline status of a feature?
+
+You can find at-a-glance Baseline statuses on [Can I Use](https://caniuse.com/) feature entries, [MDN Web Docs](https://developer.mozilla.org/) reference pages, and [webstatus.dev](https://webstatus.dev). See [Baseline in the wild](baseline-in-the-wild) for more examples.
```
On MDN and caniuse you can look up the status of individual features, but they currently don't have pages where you can see what has become available in Baseline recently. That's a use case that people ask about whenever we talk about Baseline, and that's the default sort order on the webstatus.dev start page. I think we should have a way to display that information.

jgraham replied:
If you think that use case is critical then I think linking https://web-platform-dx.github.io/web-features-explorer/ is reasonable.

petele replied:
> If you think that use case is critical then I think linking https://web-platform-dx.github.io/web-features-explorer/ is reasonable.

We do link to that on that on the "Baseline in the wild" page...

jgraham replied:
Yep. I don't mind if we also link that here, or if we leave it on the "in the wild" page. My point is that webstatus.dev doesn't have endorsement from the group and so should not be linked on the main page.

petele replied:
Removed that link.

petele commented on gh-pages/use-baseline.md:
```diff
@@ -0,0 +1,21 @@
+
+## Check if a feature works across browsers
+
+When visiting sites like [Can I Use](https://caniuse.com), [MDN Web Docs](https://developer.mozilla.org/), [RUM Archive Insights](https://rumarchive.com/insights/#baseline) and [webstatus.dev](https://webstatus.dev), check the status of a feature to learn if it works across browsers.
```
Do you have a suggestion for how to adjust this?

jgraham replied:
```suggestion
When visiting sites like [Can I Use](https://caniuse.com), [MDN Web Docs](https://developer.mozilla.org/), and [RUM Archive Insights](https://rumarchive.com/insights/#baseline) check the status of a feature to learn if it's implemented across browsers.
```

ddbeck replied:
> This tells us if it's _implemented_ across browsers, not whether it _works_. Widely available features are assumed to have good interop, but it's possible (though unfortunate) that newly available features still have notable differences.

A few points on this, @jgraham:

- The bar is higher than implemented because we ignore implementations behind flags, in future-dated releases, or using (unspecified) prefixes and aliases.
- We don't assume good interop on a timeline; the compute-baseline code actually checks what BCD reports for every release in the 30-month range and rejects releases annotated with, for example, the `partial_implementation` flag. And we do some human review. For example, I check the notes when updating features, so that (for example) BCD hasn't mistakenly allowed a partial through in a note without the requisite flag.

I'm not committed to "works" per se, but I think we can make a stronger and more accurate claim than to "implementation." Whatever the terminology, i we report a feature as Baseline and it's not working, I'd really like to hear about it. That merits an editorial override here and (very probably) a fix upstream to BCD.

jgraham replied:
"available" would also be fine.

petele replied:
Fixed in de35a122eee9e74ee11fd673eb1d4bb7c8321a7c

petele commented:
@ddbeck - As requested, I've moved files into a separate folder, created a separate package.json, etc. 
@jgraham - Removed the link as requested.

ddbeck reviewed (approved):
OK, I think this is good enough to take up incrementally now.

In my experience, something _always_ breaks with GitHub Actions, so I'm going to delay merging this until tomorrow or Monday, when I know that I have the time to do the inevitable Actions fixes and someone is around to review them. 😅

jgraham commented on gh-pages/src/index.md:
```diff
@@ -0,0 +1,38 @@
+
+## How do I find the Baseline status of a feature?
+
+You can find at-a-glance Baseline statuses on [Can I Use](https://caniuse.com/) feature entries, [MDN Web Docs](https://developer.mozilla.org/) reference pages, and [webstatus.dev](https://webstatus.dev). See [Baseline in the wild](/baseline-in-the-wild/) for more examples.
```
It looks like the webstatus.dev link didn't get removed from here as requested, but from a different page.

petele replied:
Removed it from another page - in de35a122eee9e74ee11fd673eb1d4bb7c8321a7c, and just removed it from this page in 3eea54975d50098f4c5b140b5440a5534593f60b

pepelsbey commented on gh-pages/src/_includes/default.html:
```diff
@@ -0,0 +1,67 @@
+  <head>
+    <meta charset="UTF-8">
+    <meta property="og:locale" content="en_US" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
IE in Edge mode is not relevant anymore.

```suggestion
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+    url("../fonts/Noto-Sans-700/Noto-Sans-700.woff") format("woff"),
+    url("../fonts/Noto-Sans-700/Noto-Sans-700.ttf") format("truetype"),
+    url("../fonts/Noto-Sans-700/Noto-Sans-700.svg#NotoSans") format("svg");
+  font-display: swap;
+}
+@font-face {
+  font-family: "Noto Sans";
+  font-weight: 400;
+  font-style: italic;
+  src: url("../fonts/Noto-Sans-italic/Noto-Sans-italic.eot");
+  src:
+    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.eot?#iefix")
+      format("embedded-opentype"),
+    local("Noto Sans Italic"),
+    local("Noto-Sans-italic"),
+    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.woff2") format("woff2"),
+    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.woff") format("woff"),
+    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.ttf") format("truetype"),
+    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.svg#NotoSans") format("svg");
+  font-display: swap;
+}
+@font-face {
+  font-family: "Noto Sans";
+  font-weight: 700;
+  font-style: italic;
+  src: url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.eot");
+  src:
+    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.eot?#iefix")
+      format("embedded-opentype"),
+    local("Noto Sans Bold Italic"),
+    local("Noto-Sans-700italic"),
+    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.woff2")
+      format("woff2"),
+    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.woff") format("woff"),
+    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.ttf")
+      format("truetype"),
+    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.svg#NotoSans")
+      format("svg");
+  font-display: swap;
+}
```
Removes EOT, TTF, and SVG font formats. Font files still need to be removed.

```suggestion
@font-face {
  font-family: "Noto Sans";
  font-weight: 400;
  font-style: normal;
  src:
    local("Noto Sans"),
    local("Noto-Sans-regular"),
    url("../fonts/Noto-Sans-regular/Noto-Sans-regular.woff2") format("woff2"),
    url("../fonts/Noto-Sans-regular/Noto-Sans-regular.woff") format("woff");
  font-display: swap;
}
@font-face {
  font-family: "Noto Sans";
  font-weight: 700;
  font-style: normal;
  src:
    local("Noto Sans Bold"),
    local("Noto-Sans-700"),
    url("../fonts/Noto-Sans-700/Noto-Sans-700.woff2") format("woff2"),
    url("../fonts/Noto-Sans-700/Noto-Sans-700.woff") format("woff");
  font-display: swap;
}
@font-face {
  font-family: "Noto Sans";
  font-weight: 400;
  font-style: italic;
  src:
    local("Noto Sans Italic"),
    local("Noto-Sans-italic"),
    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.woff2") format("woff2"),
    url("../fonts/Noto-Sans-italic/Noto-Sans-italic.woff") format("woff");
  font-display: swap;
}
@font-face {
  font-family: "Noto Sans";
  font-weight: 700;
  font-style: italic;
  src:
    local("Noto Sans Bold Italic"),
    local("Noto-Sans-700italic"),
    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.woff2")
      format("woff2"),
    url("../fonts/Noto-Sans-700italic/Noto-Sans-700italic.woff") format("woff");
  font-display: swap;
}
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+  -webkit-font-smoothing: subpixel-antialiased;
+}
+
+@media print, screen and (max-width: 960px) {
```
No need to specifically list `print, screen`. No media types means `all`.

```suggestion
@media (max-width: 960px) {
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+    display: inline;
+  }
+}
+@media print, screen and (max-width: 720px) {
```
```suggestion
@media (max-width: 720px) {
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+    word-wrap: normal;
+  }
+}
+@media print, screen and (max-width: 480px) {
```
```suggestion
@media (max-width: 480px) {
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+  width: 19.25em;
+  position: fixed;
+  grid-column: 1;
+  -webkit-font-smoothing: subpixel-antialiased;
```
Cross-browser font-smoothing

```suggestion
  -webkit-font-smoothing: subpixel-antialiased;
  -moz-osx-font-smoothing: grayscale;
```

pepelsbey commented on gh-pages/src/assets/css/style.css:
```diff
@@ -0,0 +1,365 @@
+  width: 19.25em;
+  position: fixed;
+  bottom: 3.5em;
+  -webkit-font-smoothing: subpixel-antialiased;
```
```suggestion
  -webkit-font-smoothing: subpixel-antialiased;
  -moz-osx-font-smoothing: grayscale;
```

pepelsbey reviewed (commented):
I’d like to suggest a few changes:

- Remove legacy code (traces of IE support)
- Some cross-browser tweaks and clean-up

Let me know if filing a separate PR later would be better.
